metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1302,6 +1302,28 @@ class MetalsLanguageServer(
               new IllegalArgumentException(s"Invalid arguments: $args.")
             )
         }).asJavaObject
+      case ServerCommands.ResolveDebugClass() =>
+        //TODO: polish this
+        val args = params.getArguments().asScala
+        val je = args.head.asInstanceOf[JsonElement]
+        val p = je.as[DebugClassParams].get
+        val (target, clazz) = (for {
+          //TODO: would be good to have error messages, maybe with use of Either
+          info <- buildTargets.findByDisplayName(
+            Option(p.project).getOrElse("root")
+          )
+          classes = buildTargetClasses.classesOf(info.getId())
+          clazz <- classes.mainClasses.values.find(
+            _.getClassName == p.mainClass
+          )
+        } yield (info.getId(), clazz)).get
+
+        val result = {
+          val dataKind = b.DebugSessionParamsDataKind.SCALA_MAIN_CLASS
+          val data = clazz.toJson
+          new b.DebugSessionParams(List(target).asJava, dataKind, data)
+        }
+        Future { result }.asJavaObject
```

Comment:
Is it possible to just run debug here instead of returning params?

Reply:
We may want to wrap the entire expression in a future to unblock future requests.

Reply:
@tgodzik ,
yeah, I'm actually thinking about maybe removing this new command and just adding another possible `Params` class(es) for `StartDebugAdapter`.
So beside `DebugSessionParams`, it could take something like
```
DebugSessionUnresolvedMainClassParams {
	mainClass: string,
	buildTarget?: string,
	args?: string[],
	jvmOptions?: string[]
}

DebugSessionUnresolvedTestClassParams {
	testClass: string,
	buildTarget?: string
}
```

also, in future for https://github.com/scalameta/metals-feature-requests/issues/81 it could process `TextDocumentPositionParams` as well.
What do you think?